### PR TITLE
BLD: Try making 64-bit Windows wheels

### PR DESCRIPTION
**Pull request by lithomas1**

### CODE REVIEW

**Comment by lithomas1** [comment]
Hopefully openblas linked correctly. This is what numpy.show_config() says.
```
 + python -c 'import numpy; numpy.show_config()'
  openblas64__info:
      library_dirs = ['C:\\Users\\runneradmin\\AppData\\Local\\Temp\\pip-req-build-1fliwbui\\build\\openblas64__info']
      libraries = ['openblas64__info']
      language = f77
      define_macros = [('HAVE_CBLAS', None), ('BLAS_SYMBOL_SUFFIX', '64_'), ('HAVE_BLAS_ILP64', None)]
  blas_ilp64_opt_info:
      library_dirs = ['C:\\Users\\runneradmin\\AppData\\Local\\Temp\\pip-req-build-1fliwbui\\build\\openblas64__info']
      libraries = ['openblas64__info']
      language = f77
      define_macros = [('HAVE_CBLAS', None), ('BLAS_SYMBOL_SUFFIX', '64_'), ('HAVE_BLAS_ILP64', None)]
  openblas64__lapack_info:
      library_dirs = ['C:\\Users\\runneradmin\\AppData\\Local\\Temp\\pip-req-build-1fliwbui\\build\\openblas64__lapack_info']
      libraries = ['openblas64__lapack_info']
      language = f77
      define_macros = [('HAVE_CBLAS', None), ('BLAS_SYMBOL_SUFFIX', '64_'), ('HAVE_BLAS_ILP64', None), ('HAVE_LAPACKE', None)]
  lapack_ilp64_opt_info:
      library_dirs = ['C:\\Users\\runneradmin\\AppData\\Local\\Temp\\pip-req-build-1fliwbui\\build\\openblas64__lapack_info']
      libraries = ['openblas64__lapack_info']
      language = f77
      define_macros = [('HAVE_CBLAS', None), ('BLAS_SYMBOL_SUFFIX', '64_'), ('HAVE_BLAS_ILP64', None), ('HAVE_LAPACKE', None)]
  Supported SIMD extensions in this NumPy install:
      baseline = SSE,SSE2,SSE3
      found = SSSE3,SSE41,POPCNT,SSE42,AVX,F16C,FMA3,AVX2,AVX512F,AVX512CD,AVX512_SKX
      not found = AVX512_CLX,AVX512_CNL,AVX512_ICL
```

File: .github/workflows/wheels.yml [hunk 1]
```diff
@@ -70,6 +70,17 @@ jobs:
         - os: macos-latest
           python: "310"
           platform: macosx_x86_64
+          
+        # Windows builds
+        - os: windows-latest
```
Comment (lithomas1) [inline .github/workflows/wheels.yml:75]: Maybe pin? Upgrading VS tends to change the DLLs required(at least from my experience with pandas), and there is no auditwheel/delocate on Windows.

**Comment by lithomas1** [comment]
Not sure if openblas linked correctly.
I see
```libraries openblas64_ not found in ['openblas', 'C:\\cibw\\python\\python.3.8.10\\tools\\lib', 'C:\\', 'C:\\cibw\\python\\python.3.8.10\\tools\\libs']```
but also ```  openblas64__info:
      library_dirs = ['C:\\Users\\runneradmin\\AppData\\Local\\Temp\\pip-req-build-r9rwumpo\\build\\openblas64__info']
      libraries = ['openblas64__info']
      language = f77
      define_macros = [('HAVE_CBLAS', None), ('BLAS_SYMBOL_SUFFIX', '64_'), ('HAVE_BLAS_ILP64', None)]
  blas_ilp64_opt_info:
      library_dirs = ['C:\\Users\\runneradmin\\AppData\\Local\\Temp\\pip-req-build-r9rwumpo\\build\\openblas64__info']
      libraries = ['openblas64__info']
      language = f77
      define_macros = [('HAVE_CBLAS', None), ('BLAS_SYMBOL_SUFFIX', '64_'), ('HAVE_BLAS_ILP64', None)]```
I think the Azure build also has the openblas not found message, so it might be fine.

**Comment by mattip** [comment]
There is one failing test, which is preventing the artifact from being uploaded. Could you xfail/skip that test for now so we can check that the artifact does indeed have openblas?

**Comment by mattip** [comment]
It is a pretty involved test anyway, mainly for testing that fortran compiling works.

**Comment by lithomas1** [comment]
> There is one failing test, which is preventing the artifact from being uploaded. Could you xfail/skip that test for now so we can check that the artifact does indeed have openblas?

Done.

**Comment by mattip** [comment]
Cool. It works:
```
>>> import ctypes
>>> dll = ctypes.CDLL(r"C:\temp\vtest_python38\Lib\site-packages\numpy\.libs\libopenblas.EL2C6PLE4ZYW3ECEVIV3OXXGRN2NRFM2.gfortran-win_amd64.dll")
>>> get_config = dll.openblas_get_config64_
>>> get_config = dll.openblas_get_config64_
>>> get_config.restype = ctypes.c_char_p
>>> get_config()
b'OpenBLAS 0.3.18  USE64BITINT DYNAMIC_ARCH NO_AFFINITY Haswell MAX_THREADS=24'
```

**Comment by lithomas1** [comment]
This should be working now. Let me know if you want this squashed.

File: tools/wheels/cibw_test_command.sh [hunk 2]
```diff
@@ -4,10 +4,15 @@
...
 PROJECT_DIR="$1"
-UNAME="$(uname)"
 
 python -c "import numpy; numpy.show_config()"
-python -c "import sys; import numpy; sys.exit(not numpy.test('full', extra_argv=['-vv']))"
+if [[ $RUNNER_OS == "Windows" ]]; then
+    # FIXME: Don't skip tests
+    # python -c "import sys; import numpy; sys.exit(not numpy.test('full', extra_argv=['-k not test_build_import', '-v']))"
+    PY_DIR=$(python -c "import sys; print(sys.prefix)")
+    mkdir $PY_DIR/libs
```
Comment (lithomas1) [inline tools/wheels/cibw_test_command.sh:13]: The failing test was trying to write to a missing directory. ``sys.prefix`` exists but the libs folder under it doesn't. This feels very hacky, though.

Comment (mattip) [reply tools/wheels/cibw_test_command.sh:13]: Agreed. That test is a total hack.

**Comment by mattip** [comment]
LGTM. Would you like to try to add 32-bit windows or is this enough for one PR?

File: tools/wheels/cibw_test_command.sh (hunk 2 above)
Comment (mattip) [reply tools/wheels/cibw_test_command.sh:13]: Could you mention issue 20391 here so we can remove this once that is resolved?

**Comment by mattip** [comment]
Could you also add a `--std=c99` exclusion as per [this comment](https://github.com/numpy/numpy/issues/20262#issuecomment-971885803) to unbreak the macos builds

**Comment by lithomas1** [comment]
Changes to build on macOS are merged in main. I'll squash to trigger wheel builds again tomorrow.

**Comment by mattip** [comment]
The action title is Build wheel for cp${{ matrix.python }}-${{ matrix.platform } in the CI report block

**Comment by lithomas1** [comment]
> The action title is Build wheel for cp${{ matrix.python }}-${{ matrix.platform } in the CI report block

Oh, the build was getting skipped since the last commit didn't have [wheel build] in it. I've rebased and squashed again.

**Comment by mattip** [comment]
Thanks @lithomas1

**Comment by mattip** [comment]
Now only windows-32, and pypy3.8 (windows-64, osx-x86_64, manylinux2014, aarch64) are missing, but they are lower priority

**Pull request merged**